tests/Oasis/AddressTests.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "HexCoding.h"
+#include "Oasis/Address.h"
+#include "PublicKey.h"
+#include "PrivateKey.h"
+#include <gtest/gtest.h>
+#include <vector>
+
+using namespace TW;
+using namespace TW::Oasis;
+
+TEST(OasisAddress, Valid) {
+    ASSERT_TRUE(Address::isValid("oasis1qp0cnmkjl22gky6p6qeghjytt4v7dkxsrsmueweh"));
+}
+
+TEST(OasisAddress, Invalid) {
+    ASSERT_FALSE(Address::isValid("oasis1qp0cnmkjl22gky6p6qeghjytt4v7dkxsrsmuewehj"));
+    ASSERT_FALSE(Address::isValid("oasi1qp0cnmkjl22gky6p6qeghjytt4v7dkxsrsmueweh"));
+}
+
+TEST(OasisAddress, ForceInvalid) {
+    try {
+        auto addressString = "oasis1qp0cnmkjl22gky6p6qeghjytt4v7dkxsrsmuewehj";
+        auto address = Address( addressString );
+        ASSERT_EQ("This test should generate an exception as it an invalid address", "-");
+    } catch( std::invalid_argument& e1 ) {
+        ASSERT_TRUE(true);
+    } 
+}
```

Comment:
Exception checking is not correct here:
- add return statement to the catch branch, this is the positive path, ASSERT_TRUE(true) is not needed
- add following line to after the try/catch, to fail if exception is missing:
   FAIL() << "Missing expected exception";

Reply:
Thanks for the update. I made changes now.